wordpress-mobile/WordPress-Android
# Scan history UI state

Parent issue #13326

This PR adds Error and Loading states to ScanHistoryFragment (contains viewpager and fullscreen error view) and ScanHistoryListFragment (contains RecyclerView).

Overview of the Scan History screen structure
1. ScanHistory (parent) - takes care of fetching the data from the server and showing an error when the request fails. It provides LiveData with a list of all threats.
2. ScanHistoryList (child, one per tab) - observes threats provided by ScanHistoryViewModel, filters them by ThreatStatus depending on the corresponding tab, and maps them to UiState. Takes care of showing list of skeleton items (loading), list of threats or empty screen when the list of threats is empty.


To test:
1. Make sure Scan feature flag is enabled
2. Open Scan
3. Click on History action in the toolbar (notice loading skeletons)
4. Turn on airplane mode before the request finishes
5. Notice a RequestFailed error is shown
7. Click on Retry
8. Notice NoConnection error is shown
9. Turn the airplane mode off
10. Click on retry
11. Notice threat items are shown when the request finishes (it might take 30s)
-----------------
Switch to a site which doesn't have any threats in history
1. Open Scan
2. Click on History action in the toolbar
3. Notice empty screen is shown when the request finishes (it might take 30s)

PR submission checklist:
